google/cloud/storage/client_options_test.cc

```diff
@@ -445,20 +445,21 @@ TEST_F(ClientOptionsTest, ProjecIdtWithEnv) {
   EXPECT_EQ("my-project", options.get<ProjectIdOption>());
 }
 
-TEST_F(ClientOptionsTest, UseRestClientWithoutEnv) {
-  ScopedEnvironment env("GOOGLE_CLOUD_CPP_STORAGE_HAVE_REST_CLIENT",
+TEST_F(ClientOptionsTest, UseRestClientByDefault) {
+  ScopedEnvironment env("GOOGLE_CLOUD_CPP_STORAGE_USE_LEGACY_HTTP",
                         absl::nullopt);
   auto const options =
       internal::DefaultOptions(oauth2::CreateAnonymousCredentials(), {});
-  EXPECT_FALSE(options.has<internal::UseRestClientOption>());
+  EXPECT_TRUE(options.has<internal::UseRestClientOption>());
```

Comment:
Should `UseRestClientOption` become public?  If not, why document it in the CHANGELOG?